NuRadioMC/SignalProp/analyticraytracing.py

```diff
@@ -1744,7 +1749,362 @@ def get_path(self, iS, n_points=1000):
+        nz: float
+            the index of refraciton into the z direction         
+        rounding: int
+            the precision of the comparison between n and nx, ny, nz for the special cases    
+        """ 
+        
+        if(np.round(n, rounding) in [np.round(nx, rounding), np.round(ny, rounding), np.round(nz, rounding)]):
+        # treat special cases here
+            if(np.isclose(direction[0], 1)):  # propagation into x
+                print(f"propagating into x")
+                if(np.isclose(n, ny)):
+                    return np.array([0, 1, 0])
+                elif(np.isclose(n, nz)):
+                    return np.array([0, 0, 1])
+            elif(np.isclose(direction[1], 1)):  # propagation into y
+                print(f"propagating into y")
+                if(np.isclose(n, nx)):
+                    return np.array([1, 0, 0])
+                elif(np.isclose(n, nz)):
+                    return np.array([0, 0, 1])
+            elif(np.isclose(direction[2], 1)):  # propagation into z
+                print(f"propagating into z")
+                if(np.isclose(n, nx)):
+                    return np.array([1, 0, 0])
+                elif(np.isclose(n, ny)):
+                    return np.array([0, 1, 0])
+        # now treat the cases where the direction is perpendicular to just one of the principal axes
+            elif(np.isclose(np.dot(direction, np.array([1, 0, 0])), 0)):  # direction perpendicular to x-axis
+                if(np.isclose(n, nx)):
+                    return np.array([1, 0, 0])
+                else:
+                    print(f"ERROR: propagation direction perpendicular to x-axis but n not equal to n_x {n}")
+                    return [0, 0, 0]
+            elif(np.isclose(np.dot(direction, np.array([0, 1, 0])), 0)):  # direction perpendicular to y-axis
+                if(np.isclose(n, ny)):
+                    return np.array([0, 1, 0])
+                else:
+                    print(f"ERROR: propagation direction perpendicular to y-axis but n not equal to n_y")
+                    return [0, 0, 0]
+            elif(np.isclose(np.dot(direction, np.array([0, 0, 1])), 0)):  # direction perpendicular to z-axis
+                if(np.isclose(n, nz)):
+                    return np.array([0, 0, 1])
+                else:
+                    print(f"ERROR: propagation direction perpendicular to z-axis but n not equal to n_z")
+                    return [0, 0, 0]
+
+            else:
+                print(f"ERROR: non of the special cases worked")
+                print(n, direction, nx, ny, nz)
+                return [0, 0, 0]
+        else:
+            p = np.array([direction[0] / (n ** 2 - nx ** 2), direction[1] / (n ** 2 - ny ** 2), direction[2] / (n ** 2 - nz ** 2)])
+            return p / np.linalg.norm(p)
+
+    def get_3d_trace(self, source, antenna, acc = 1000):
+        
+        """
+        returns the trace of the two possible solutions provided by get_path
```

Comment:
Please document the format of the output

Reply:
I also will not comment on future functions. I am sensing a theme that this is missing in many cases.